src/renderer/components/kube-object-menu/kube-object-menu-items.injectable.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable, lifecycleEnum } from "@ogre-tools/injectable";
+
+import type { KubeObject } from "../../../common/k8s-api/kube-object";
+import { computed } from "mobx";
+import { computedInjectManyInjectable } from "@ogre-tools/injectable-extension-for-mobx";
+import { kubeObjectMenuItemInjectionToken } from "./kube-object-menu-item-injection-token";
+import { filter, map, sortBy } from "lodash/fp";
+import { pipeline } from "@ogre-tools/fp";
+
+const kubeObjectMenuItemsInjectable = getInjectable({
+  id: "kube-object-menu-items",
+
+  instantiate: (di, { kubeObject }: { kubeObject: KubeObject }) => {
+    const computedInjectMany = di.inject(computedInjectManyInjectable);
+    const menuItems = computedInjectMany(kubeObjectMenuItemInjectionToken);
+
+    return computed(() =>
+      pipeline(
+        menuItems.get(),
+
+        filter(
+          (item) =>
+            item.kind === kubeObject?.kind &&
+            item.apiVersions.includes(kubeObject?.apiVersion) &&
+            item.enabled.get(),
+        ),
+
+        sortBy((item) => item.orderNumber),
+        map((item) => item.Component),
+      ),
+    );
+  },
+
+  lifecycle: lifecycleEnum.keyedSingleton({
+    getInstanceKey: (di, { kubeObject }: { kubeObject: KubeObject }) =>
```

Comment:
shouldn't it be `KubeObject | undefined`?

Reply:
Hmm. In here it looks like it yeah, but from the perspective of usage (kube-object-menu) it doesn't. I'd would like to be sure it's not undefined.